javascript/src/main/pages/Home/Home.js

```diff
@@ -3,17 +3,32 @@ import { Jumbotron } from "react-bootstrap";
 import CourseTable from "main/components/Courses/CourseTable"
 import useSWR from "swr";
 import { fetchWithoutToken } from "main/utils/fetch";
+import { useHistory } from "react-router-dom";
 
 const Home = () => {
     const { data: courses } = useSWR(
         "/api/public/courses",
         fetchWithoutToken
     );
+    const { data: currentFilter } = useSWR(
+        "api/public/filter", fetchWithoutToken
+    );
+    const filterLabel = (currentFilter) =>{
+        if(currentFilter && currentFilter.length > 0){
+            if(currentFilter[0].activeQuarter !=="All"){
```

Comment:
Maybe lift these into the same if statement?

Reply:
Gave that a try, but when I do that, it says currentfilter is not defined since I'm waiting on the Url to respond

Reply:
If statements are short circuiting, so if you make this the last statement I would think it _wouldn't_ do that. Did you try that?

Reply:
Did you make it the last check in the list of booleans? JavaScript is short-circuiting, so that should solve the problem.